### ISSUE TITLE
Kolibri version issue

### ISSUE THREAD

**Issue by christianmemije** [learningequality/kolibri-kivy#11]
Attempting to build using a whl file from Develop or 0.5.

```
Traceback (most recent call last):
--
  | File "/usr/local/bin/buildozer", line 11, in <module>
  | sys.exit(main())
  | File "/usr/local/lib/python2.7/dist-packages/buildozer/scripts/client.py", line 13, in main
  | Buildozer().run_command(sys.argv[1:])
  | File "/usr/local/lib/python2.7/dist-packages/buildozer/__init__.py", line 1058, in run_command
  | self.target.run_commands(args)
  | File "/usr/local/lib/python2.7/dist-packages/buildozer/target.py", line 91, in run_commands
  | func(args)
  | File "/usr/local/lib/python2.7/dist-packages/buildozer/target.py", line 103, in cmd_debug
  | self.buildozer.build()
  | File "/usr/local/lib/python2.7/dist-packages/buildozer/__init__.py", line 212, in build
  | self.target.build_package()
  | File "/usr/local/lib/python2.7/dist-packages/buildozer/targets/android.py", line 642, in build_package
  | version = self.buildozer.get_version()
  | File "/usr/local/lib/python2.7/dist-packages/buildozer/__init__.py", line 717, in get_version
  | ' (looking for `{1}`)'.format(fn, regex))
  | Exception: Unable to find capture version in ./code/kolibri/VERSION
  | (looking for `v(\d+\.\d+\.\d+)`)
```

```
08-17 11:04:30.954 31390 31451 I python  : Traceback (most recent call last):
08-17 11:04:30.954 31390 31451 I python  :   File "main.py", line 11, in <module>
08-17 11:04:30.954 31390 31451 I python  :     django.setup()
08-17 11:04:30.954 31390 31451 I python  :   File "kolibri/dist/django/__init__.py", line 17, in setup
08-17 11:04:30.954 31390 31451 I python  :     configure_logging(settings.LOGGING_CONFIG, settings.LOGGING)
08-17 11:04:30.954 31390 31451 I python  :   File "kolibri/dist/django/conf/__init__.py", line 55, in __getattr__
08-17 11:04:30.954 31390 31451 I python  :     self._setup(name)
08-17 11:04:30.954 31390 31451 I python  :   File "kolibri/dist/django/conf/__init__.py", line 43, in _setup
08-17 11:04:30.954 31390 31451 I python  :     self._wrapped = Settings(settings_module)
08-17 11:04:30.954 31390 31451 I python  :   File "kolibri/dist/django/conf/__init__.py", line 99, in __init__
08-17 11:04:30.954 31390 31451 I python  :     mod = importlib.import_module(self.SETTINGS_MODULE)
08-17 11:04:30.954 31390 31451 I python  :   File "/data/user/0/org.le.kolibri/files/lib/python27.zip/importlib/__init__.py", line 37, in import_module
08-17 11:04:30.954 31390 31451 I python  :     
08-17 11:04:30.954 31390 31451 I python  :   File "/data/user/0/org.le.kolibri/files/kolibri/__init__.py", line 13, in <module>
08-17 11:04:30.954 31390 31451 I python  :     __version__ = str(get_version(VERSION))
08-17 11:04:30.954 31390 31451 I python  :   File "/data/user/0/org.le.kolibri/files/kolibri/utils/lru_cache.py", line 137, in wrapper
08-17 11:04:30.954 31390 31451 I python  :     result = user_function(*args, **kwds)
08-17 11:04:30.954 31390 31451 I python  :   File "/data/user/0/org.le.kolibri/files/kolibri/utils/version.py", line 404, in get_version
08-17 11:04:30.955 31390 31451 I python  :     return get_prerelease_version(version)
08-17 11:04:30.955 31390 31451 I python  :   File "/data/user/0/org.le.kolibri/files/kolibri/utils/version.py", line 332, in get_prerelease_version
08-17 11:04:30.955 31390 31451 I python  :     tag_describe = get_git_describe()
08-17 11:04:30.955 31390 31451 I python  :   File "/data/user/0/org.le.kolibri/files/kolibri/utils/version.py", line 237, in get_git_describe
08-17 11:04:30.955 31390 31451 I python  :     universal_newlines=True
08-17 11:04:30.955 31390 31451 I python  :   File "/home/christian/repos/kolibri-kivy/.buildozer/android/platform/build/build/other_builds/python2-openssl-sqlite3/armeabi-v7a/python2/python-install/lib/python2.7/subprocess.py", line 679, in __init__
08-17 11:04:30.955 31390 31451 I python  :   File "/home/christian/repos/kolibri-kivy/.buildozer/android/platform/build/build/other_builds/python2-openssl-sqlite3/armeabi-v7a/python2/python-install/lib/python2.7/subprocess.py", line 1228, in _execute_child
08-17 11:04:30.955 31390 31451 I python  : OSError: [Errno 2] No such file or directory
08-17 11:04:30.996 31390 31451 I python  : Python for android ended.

```

**Comment by aronasorman** [comment]
Looks like `subprocess` doesn't exist in the version of the python interpreter you're pulling in @jamalex @christianmemije. We'll have to add a try-except clause to skip running `git describe` if `subprocess` doesn't exist.

**Comment by jamalex** [comment]
What we need is to be able to get the current Kolibri version from the extracted WHL file. We used to get that from the `VERSION` file, but it seems to now be gone.

`Exception: Unable to find capture version in ./code/kolibri/VERSION`

**Comment by aronasorman** [comment]
Are you able to run python code before this? If so, you can just execute `python -c "import kolibri; kolibri.__version__` but there might be a cleaner version (@benjaoming any recommendations?)

**Comment by jamalex** [comment]
Not as part of the buildozer config (it takes a file and a regex for extracting the version number, which could perhaps work also?), but we could do that as part of the docker file, if needed.

**Comment by christianmemije** [comment]
> We used to get that from the VERSION file, but it seems to now be gone.

It is still there, the regex just needs to be updated since the version schema changed.

**Comment by christianmemije** [comment]
Python for Android expects only digits and decimals in the version.

https://github.com/kivy/python-for-android/blob/395c7e0eb5709df316fc04bab00f8c887ae40ea2/pythonforandroid/bootstraps/webview/build/build.py#L283

For something like `0.6.dev020170912184629-git`, we can either extract just `0.6`, or can you think of a way to match all digits and decimals as one match?

**Comment by christianmemije** [comment]
Looks like subprocess doesn't exist in the version of the python interpreter you're pulling in

> I think the actual issue it that git is not installed.

**Comment by christianmemije** [comment]
Fixed in #15
